src/troute-network/troute/AbstractNetwork.py

```diff
@@ -64,25 +47,142 @@ def __init__(self, cols=None, terminal_code=None, break_points=None, verbose=Fal
             dtype="float32",
         )
         """
-        # there may be off-domain nodes that are not explicitly identified
-        # but which are terminal (i.e., off-domain) as a result of a mask or some other
-        # an interior domain truncation that results in a
-        # otherwise valid node value being pointed to, but which is masked out or
-        # being intentionally separated into another domain.
-        self._terminal_codes = set(
-            self._dataframe[
-                ~self._dataframe["downstream"].isin(self._dataframe.index)
-            ]["downstream"].values
-        )
-         # There can be an externally determined terminal code -- that's this value
-        self._terminal_codes.add(terminal_code)
-
         self._break_segments = set()
-        if break_points:
-            if break_points["break_network_at_waterbodies"]:
+        
+        if self.break_points:
```

Comment:
Yes, but for no reason in particular. In fact, the code for doing this in HYFeatures and NHDNetwork is identical so it could just be moved into the superclass.

I suppose if we think break_points might be something different for future network objects we could establish this as an `@abtractmethod`, but for now moving it into the superclass is easiest.

Reply:
Moved creation of self.break_points into superclass